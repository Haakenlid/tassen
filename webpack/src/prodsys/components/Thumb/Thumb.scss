// Thumbnail image centered and contained in flexbox
@import 'styles/globals.scss';
.Thumb {
  display: flex;
  min-height: 0; // firefox
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: rgba(black, 0.15);
  .CropBoxContainer,
  img {
    flex: 1 1 0;
    object-fit: contain;
    max-height: 100%;
  }
  &.dropshadow {
    // faximile
    background: none;
    img {
      flex: 0 0 auto;
      max-width: calc(100% - #{2 * $margin});
      max-height: calc(100% - #{2 * $margin});
      box-shadow: 0.2em 0.4em 0.5em rgba(black, 0.3);
    }
  }
}
