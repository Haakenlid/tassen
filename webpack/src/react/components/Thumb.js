import React from 'react'
import cx from 'classnames'
import 'styles/thumb.scss'

const Thumb = ({ className, ...props }) => (
  <div className={cx('Thumb', className)}>
    <img {...props} />
  </div>
)
export default Thumb
