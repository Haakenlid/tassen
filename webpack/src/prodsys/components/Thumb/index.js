import cx from 'classnames'
import './Thumb.scss'

const Thumb = ({ className, ...props }) => (
  <div className={cx('Thumb', className)}>
    <img {...props} />
  </div>
)
export default Thumb
