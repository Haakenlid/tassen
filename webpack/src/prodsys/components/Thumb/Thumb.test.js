import Thumb from './index'
import renderer from 'react-test-renderer'

describe('Thumb', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<Thumb />).toJSON()
    expect(tree).toEqual(expect.any(Object))
    expect(tree).toMatchSnapshot()
  })
})
