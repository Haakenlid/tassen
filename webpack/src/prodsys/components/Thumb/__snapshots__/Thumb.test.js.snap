// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumb renders correctly 1`] = `
<div
  className="Thumb"
>
  <img />
</div>
`;
